### ISSUE TITLE
MacOS release crashing on start-up 

### ISSUE THREAD

**Issue by projx** [njust/KTail#3]
Hi 

Just a heads up, the app crashes on MacOS 12.x, it doesn't even show the UI.. I've pulled the following information out of the Console logs - It seems an issue with "libffi.8.dylib" missing, so I installed it with brew (brew install libftdi).  But the error keeps re-occuring:

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: /usr/local/opt/libffi/lib/libffi.8.dylib
Referenced from: /private/var/folders/*/KTail.app/Contents/MacOS/lib/libgobject-2.0.0.dylib
Reason: tried: '/private/var/folders/6l/04g16thj6pb68vb3qmv0yksm0000gn/T/AppTranslocation/BB746C95-00E9-4ED5-85C6-C153FB136D08/d/KTail.app/Contents/MacOS/lib/libffi.8.dylib' (code signature invalid (errno=1) sliceOffset=0x00000000, codeBlobOffset=0x00008610, codeBlobSize=0x00004890 for '/private/var/folders/6l/04g16thj6pb68vb3qmv0yksm0000gn/T/AppTranslocation/BB746C95-00E9-4ED5-85C6-C153FB136D08/d/KTail.app/Contents/MacOS/lib/libffi.8.dylib'), '/usr/local/opt/libffi/lib/libffi.8.dylib' (no such file), '/usr/local/lib/libffi.8.dylib' (no such file), '/usr/lib/libffi.8.dylib' (no such file)
(terminated at launch; ignore backtrace)

Application Specific Information:
Library not loaded: /usr/local/opt/libffi/lib/libffi.8.dylib
Referenced from: /private/var/folders/6l/04g16thj6pb68vb3qmv0yksm0000gn/T/AppTranslocation/BB746C95-00E9-4ED5-85C6-C153FB136D08/d/KTail.app/Contents/MacOS/lib/libgobject-2.0.0.dylib
Reason: tried: '/private/var/folders/6l/04g16thj6pb68vb3qmv0yksm0000gn/T/AppTranslocation/BB746C95-00E9-4ED5-85C6-C153FB136D08/d/KTail.app/Contents/MacOS/lib/libffi.8.dylib' (code signature invalid (errno=1) sliceOffset=0x00000000, codeBlobOffset=0x00008610, codeBlobSize=0x00004890 for '/private/var/folders/6l/04g16thj6pb68vb3qmv0yksm0000gn/T/AppTranslocation/BB746C95-00E9-4ED5-85C6-C153FB136D08/d/KTail.app/Contents/MacOS/lib/libffi.8.dylib'), '/usr/local/opt/libffi/lib/libffi.8.dylib' (no such file), '/usr/local/lib/libffi.8.dylib' (no such file), '/usr/lib/libffi.8.dylib' (no such file)

If it helps, I'm on a Macbook Air M1..

**Comment by njust** [comment]
Hi, thanks for reporting this. Unfortunately, my macOS knowledge is limited. I don't own a Mac. I tested KTail on a virtual machine with macOS Catalina, which worked quite fine. 

The libffi8.dylib exists in the lib directory of KTail here "KTail.app/Contents/MacOS/lib". The error says that the code signature is invalid. That's probably because the binaries are compiled for x86 / x64, or a different version of libffi is needed. 
Assuming that rosetta should handle the x64 emulation, probably the latter is the case. 

As far as I understand, "/usr/lib/libffi.dylib" should exist and "libffi.8.dylib" should point to it. I updated the macOS build accordingly and released v0.4.1. Hopefully it works for you.
